<a href="https://colab.research.google.com/github/Satgrays/ActividadFormativaM2-Equipo6/blob/main/M2_1_ManzurMaciasPineda.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Actividad M2.1 Google Colab

### Manzur Macias Pineda - A01198234

### Erandi Abigail Ramírez Muñoz A01656031

### Ángel David Ávila Pérez A01562833

### Jesús Marco Muñoz Hernández A01721819

### Felipe de Jesús Felix Arredondo A00833150

### Andrés Agustín Martínez Girón A00834798



In [ ]:
import pandas as pd

# Intentando agregar mas cosas desde Colab

I just came to say hello, hello 

I'm not the kinda girl to get messed up with you, hello

F. Agus

## Lectura de Datos

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
cd "/content/drive/MyDrive/Análisis de ciencia de datos/M2 Actividad 1"

/content/drive/MyDrive/Análisis de ciencia de datos/M2 Actividad 1


In [ ]:
titanic = pd.read_csv("titanic.csv")

In [ ]:
titanic.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [ ]:
titanic.tail(5)

PassengerId  Survived  Pclass                                      Name  \
886          887         0       2                     Montvila, Rev. Juozas   
887          888         1       1              Graham, Miss. Margaret Edith   
888          889         0       3  Johnston, Miss. Catherine Helen "Carrie"   
889          890         1       1                     Behr, Mr. Karl Howell   
890          891         0       3                       Dooley, Mr. Patrick   

        Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
886    male  27.0      0      0      211536  13.00   NaN        S  
887  female  19.0      0      0      112053  30.00   B42        S  
888  female   NaN      1      2  W./C. 6607  23.45   NaN        S  
889    male  26.0      0      0      111369  30.00  C148        C  
890    male  32.0      0      0      370376   7.75   NaN        Q

## Descripción de Variables

In [ ]:
titanic.dtypes

PassengerId      int64
Survived         int64
Pclass           int64
Name            object
Sex             object
Age            float64
SibSp            int64
Parch            int64
Ticket          object
Fare           float64
Cabin           object
Embarked        object
dtype: object

Se encuentran dentro de la base de datos 2 tipos de variables, las categóricas que generalmente estan llenadas por strings or objects y las numéricas que generalmente son integers or floats.

Variables Categóricas
*   Name
*   Sex
*   Ticket
*   Cabin
*   Embarked

Variables Numéricas
*   PassengerId
*   Survived
*   Pclass
*   Age
*   SibSp
*   Parch
*   Fare



In [ ]:
titanic.columns

Index(['PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp',
       'Parch', 'Ticket', 'Fare', 'Cabin', 'Embarked'],
      dtype='object')

 la función columns muestra únicamente el nombre de todas las columnas a manera de string con las que cuenta la base de datos seleccionada.

In [ ]:
titanic.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


Lo que hace la función info() es mostrar el nombre de todas las columnas con las que cuenta la base de datos seleccionada, a diferencia de la funcion columns además de mostrar los nombres de las columnas, también muestra el número de la columna, el conteo total de datos que no son de tipo NULL, el tipo de datos que alberga esa columna, el número de filas y columnas total en la base de datos, la clase de la base de datos y finalmente la cantidad de memoria usada en Kilobites.

In [ ]:
titanic.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

Y por ultimo, la función describe() muestra diferentes conceptos estadísticos aplicados a cada una de las columnas numéricas de la base de datos. Por ejemplo, se puede observar que para cada columna numérica, se tiene el conteo de datos, el promedio, la desviación estándar, el valor mínimo, los cuartíles (25%, 50% y 75%) y finalmente el valor máximo.

Por lo que, las columnas que aparecen en el resultado son aquellas columnas que son de tipo numérico. Siendo así la tarifa más alta de 512.329200 y el promedio de la edad es de 29.699118.

In [ ]:
len(titanic)

891

Se tienen en total 891 registros (filas) en toda la base de datos de titanic.

In [ ]:
titanic.isna().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

Donde se observa que las únicas columnas que contienen valores nulos son 'Age', 'Cabin' y 'Embarked'. Teniendo así, 177 valores nulos para 'Age', 687 valores nulos para 'Cabin' y 2 valores nulos para 'Embarked'.

Las demás columnas que no contienen valores nulos son: 'PassengerId', 'Survived', 'Pclass', 'Name', 'Sex', 'SibSp', 'Parch', 'Ticket' y 'Fare'.

In [ ]:
print("La cantidad de variables diferentes en la columna 'Name' es de: ", len(titanic['Name'].unique()))

La cantidad de variables diferentes en la columna 'Name' es de:  891


In [ ]:
print("La cantidad de variables diferentes en la columna 'Sex' es de: ", len(titanic['Sex'].unique()))

La cantidad de variables diferentes en la columna 'Sex' es de:  2


In [ ]:
print("La cantidad de variables diferentes en la columna 'Ticket' es de: ", len(titanic['Ticket'].unique()))

La cantidad de variables diferentes en la columna 'Ticket' es de:  681


In [ ]:
print("La cantidad de variables diferentes en la columna 'Cabin' es de: ", len(titanic['Cabin'].unique()))

La cantidad de variables diferentes en la columna 'Cabin' es de:  148


In [ ]:
print("La cantidad de variables diferentes en la columna 'Embarked' es de: ", len(titanic['Embarked'].unique()))

La cantidad de variables diferentes en la columna 'Embarked' es de:  4


## Análisis de Datos

In [ ]:
print("La suma del total que se pagó sumando la tarifa de todos los pasajeros es de: ", titanic['Fare'].sum())

La suma del total que se pagó sumando la tarifa de todos los pasajeros es de:  28693.9493


In [ ]:
titanic_survived = titanic[titanic['Survived'] == 1]
titanic_survived.head()

PassengerId  Survived  Pclass  \
1            2         1       1   
2            3         1       3   
3            4         1       1   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

In [ ]:
titanic_dead = titanic[titanic['Survived'] == 0]
titanic_dead.head()

PassengerId  Survived  Pclass                            Name   Sex   Age  \
0            1         0       3         Braund, Mr. Owen Harris  male  22.0   
4            5         0       3        Allen, Mr. William Henry  male  35.0   
5            6         0       3                Moran, Mr. James  male   NaN   
6            7         0       1         McCarthy, Mr. Timothy J  male  54.0   
7            8         0       3  Palsson, Master. Gosta Leonard  male   2.0   

   SibSp  Parch     Ticket     Fare Cabin Embarked  
0      1      0  A/5 21171   7.2500   NaN        S  
4      0      0     373450   8.0500   NaN        S  
5      0      0     330877   8.4583   NaN        Q  
6      0      0      17463  51.8625   E46        S  
7      3      1     349909  21.0750   NaN        S